src/igvfd/audit/file_set.py

```diff
@@ -137,27 +157,75 @@ def audit_unexpected_seqspec(value, system):
+                    has_seqspec_document.append(file)
+
+        # Audit message for Audit 1
         if has_seqspec:
-            has_seqspec = ', '.join([audit_link(path_to_text(file_with_seqspec), file_with_seqspec)
-                                     for file_with_seqspec in has_seqspec])
+            audit_message_pod5 = get_audit_message(audit_unexpected_seqspec, index=0)
+            has_seqspec_paths = join_obj_paths(data_object_paths=has_seqspec)
             detail = (
                 f'{object_type} {audit_link(path_to_text(value["@id"]), value["@id"])} has pod5 sequence file(s): '
-                f'{has_seqspec} which are linked to a seqspec.'
+                f'{has_seqspec_paths} which are linked to a seqspec.'
             )
-            yield AuditFailure(audit_message.get('audit_category', ''), f'{detail} {audit_message.get("audit_description", "")}', level=audit_message.get('audit_level', ''))
+            yield AuditFailure(audit_message_pod5.get('audit_category', ''), f'{detail} {audit_message_pod5.get("audit_description", "")}', level=audit_message_pod5.get('audit_level', ''))
+
+        # Audit message for Audit 2
+        if had_duo_seqspecs:
+            audit_msg_double_seqspec = get_audit_message(audit_unexpected_seqspec, index=1)
+            had_duo_seqspecs_paths = join_obj_paths(data_object_paths=had_duo_seqspecs)
+            detail = (
+                f'{object_type} {audit_link(path_to_text(value["@id"]), value["@id"])} has sequence file(s): '
+                f'{had_duo_seqspecs_paths} which are linked to both sequence specification files as Configuration File and library structure seqspec as Document.'
+            )
+            yield AuditFailure(audit_msg_double_seqspec.get('audit_category', ''), f'{detail} {audit_msg_double_seqspec.get("audit_description", "")}', level=audit_msg_double_seqspec.get('audit_level', ''))
+
+        # Audit message for Audit 3
+        if has_seqspec_document:
+            wrong_documents = []
+            for seqfile in has_seqspec_document:
+                # Get file obj to get to seqspec doc
+                seqfile_obj = system.get('request').embed(seqfile + '@@object?skip_calculated=true')
+                # Get to document obj to get doc type
+                document_obj = system.get('request').embed(seqfile_obj.get(
+                    'seqspec_document') + '@@object?skip_calculated=true')
+                if document_obj.get('document_type') != 'library structure seqspec':
+                    wrong_documents.append(seqfile)
+            audit_msg_wrong_seqspec_doc = get_audit_message(audit_unexpected_seqspec, index=2)
+            wrong_document_paths = join_obj_paths(data_object_paths=wrong_documents)
+            detail = (
+                f'{object_type} {audit_link(path_to_text(value["@id"]), value["@id"])} has sequence file(s): '
+                f'{wrong_document_paths} which are linked to Document.'
```

Comment:
which link to a seqspec_document that is not a library structure seqspec document?